### DISCUSSION TITLE
Manim in Ruby?

### DISCUSSION THREAD

**Discussion by andydevs** [ManimCommunity/manim#2666] [General]
This is just a quick idea or discussion. I feel like writing Manim in Ruby (or even a Ruby wrapper for manim), might be cool. I think ruby's syntax can make coding for manim pretty useful.

I'm imagining something like this scene

```py
class MyScene(Scene):
    def construct(self):
        circle = Circle()
        circle.set_fill(BLUE, opacity=0.5)

        self.play(ShowCreation(circle))
        self.wait()

        square = Square()
        square.set_stroke(BLUE_E)
        self.play(Transform(circle, square))
```

Could be written like this

```ruby
Scene.construct "MyScene" do
    circle = Circle.new
    circle.set_fill BLUE, opacity:0.5
   
    play Animation.show_creation(circle)
    wait

    square = Square.new
    square.set_stroke BLUE_E
    play Animation.transform circle, square
end
```

That being said, I understand that Python is a widely more accessible language than Ruby, which makes it the obvious choice for writing an animation program that is accessible to the most people.

I guess I want to pose the question, what do you think of Ruby? What if one were to, per say, rewrite manim or create another animation system like manim using Ruby (hypothetically)? Or there other advantages to using python other than it's popularity?

**Comment by Darylgolden** [comment]
While it would be cool to have Manim in Ruby, I don't think the added benefit is worth the effort required to develop and maintain a Ruby clone/wrapper. If you wish to write an animation library for Ruby, you may as well write something new, avoiding many of the inherent flaws of Manim's current design.

**Reply by andydevs** [reply]
So I'm coming back to this chat after a while.

I'm curious, what in your opinion are the current issues with Manim's design
